transport/src/main/java/me/uport/sdk/transport/ResponseParser.kt

```diff
@@ -115,41 +118,60 @@ object ResponseParser {
     fun extractTokenFromIntent(intent: Intent?): UriResponse {
         intent ?: throw IllegalArgumentException("Can't process a null intent")
 
+        ActivityInfo.LAUNCH_SINGLE_TASK
         val appLinkData: Uri? = intent.data
                 ?: throw IllegalArgumentException("Can't process an intent with no data")
 
         if (Intent.ACTION_VIEW == intent.action) {
             return extractTokenFromRedirectUri(appLinkData.toString())
-        }
-        else {
+        } else {
             throw IllegalArgumentException("Intent action has to be ${Intent.ACTION_VIEW}")
         }
     }
+
+    /**
+     * Use this method to extract a response token from a deep-link callback.
+     * @returns a [UriResponse] when the response contains one and the token can be parsed or `null`
+     * when the action was canceled or the response does not belong to the uPort domain
+     */
+    fun parseActivityResult(requestCode: Int, resultCode: Int, data: Intent?): UriResponse? {
+        if (resultCode == Activity.RESULT_CANCELED || requestCode != Transports.UPORT_DEFAULT_REQUEST_CODE) {
+            return null
+        }
+        val redirectUri = data?.getStringExtra(RequestDispatchActivity.EXTRA_REDIRECT_URI) ?: ""
+        return try {
+            ResponseParser.extractTokenFromRedirectUri(redirectUri)
+        } catch (err: IllegalArgumentException) {
+            null
+        }
+
+    }
 }
 
 /**
  * Generic class for handling various response types
  **
  */
-sealed class UriResponse
+sealed class UriResponse {
 
+    /**
+     * Data Class to handle all JWT response types
+     **
+     */
+    class JWTUriResponse(val token: String) : UriResponse()
```

Comment:
Any reason for making there nested classes?

Reply:
It's the most common pattern with sealed classes.
Now that you raised the question, it seems kotlin is more relaxed about this pattern since v1.1 so I'm going to revert this. There will be an added bonus that the pattern can be switched to interfaces instead of sealed class.

Reply:
Fixed it. Please check the latest changes